dds/DCPS/TopicDetails.h

```diff
@@ -105,9 +110,76 @@ namespace OpenDDS {
         endpoints_.insert(guid);
       }
 
+      // Remote (assumes type_ids != TK_NONE)
+      void add_pub_sub_xtype(const DCPS::RepoId& remote_guid,
+                       const XTypes::TypeIdentifier& local_type_id,
+                       const XTypes::TypeIdentifier& remote_type_id)
+      {
+
+        RepoId participant_id = remote_guid;
+        participant_id.entityId = ENTITYID_PARTICIPANT;
+
+        endpoints_.insert(remote_guid);
+
+        RemoteTopicMap::iterator remote_topic_iter = remote_topics_.find(participant_id);
+        bool inconsistent_before;
+        if (remote_topic_iter == remote_topics_.end()) {
+          remote_topic_iter = remote_topics_.insert(std::make_pair(participant_id, RemoteTopic())).first;
+          inconsistent_before = false;
+        } else {
+          inconsistent_before = remote_topic_iter->second.inconsistent_;
+        }
+
+        // Initialize.
+        RemoteTopic& remote = remote_topic_iter->second;
+        remote.data_type_name_.clear();
+        remote.type_id_= remote_type_id;
+        XTypes::TypeAssignability ta(make_rch<XTypes::TypeLookupService>());
+        remote.inconsistent_ = topic_callbacks_ && ta.assignable(local_type_id, remote.type_id_);
```

Comment:
Looks like this would involve `!ta.assignable(local_type_id, remote.type_id)` on the right-hand side instead because it is checking inconsistency.

Reply:
good point.